src/App.tsx

```diff
@@ -15,47 +16,130 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return SortType.NONE;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export const App: React.FC<{}> = () => {
+  const [firstState, secondState] = useState<State>({
```

Comment:
you should change names a little bit according to naming useState variables 

Examples: 
```
const [sortType, setSortType] = useState('')
const [isValid, setIsValid] = useState(true)
```

Reply:
yeah, naming is the hardest. )))